Tasks/VsTest/vstest.ts

```diff
@@ -0,0 +1,515 @@
+                        defer.resolve(settingsFile);
+                    });
+            }
+            else {
+                tl.debug("Adding maxcpucount element to runsettings file provided.");
+                readFileContents(settingsFile, "utf-8").then(function(xmlContents) {
+                    var parser = new xml2js.Parser();
+                    parser.parseString(xmlContents, function(err, result) {
+                        if (err) {
+                            tl.warning("Error occured while reading run settings file. Error : " + err);
+                            tl.debug("Error occured while setting run in parallel. Continuing...");
+                            defer.resolve(settingsFile);
+                        }
+                        else {
+                            if (result.RunSettings === undefined) {
+                                tl.warning("Failed to set run in parallel. Invalid run settings file.");
+                                defer.resolve(settingsFile);
+                            }
+                            else if (!result.RunSettings) {
+                                result.RunSettings = { RunConfiguration: { MaxCpuCount: 0 } };
+                            }
+                            else if (!result.RunSettings.RunConfiguration || !result.RunSettings.RunConfiguration[0]) {
+                                result.RunSettings.RunConfiguration = { MaxCpuCount: 0 };
+                            }
+                            else {
+                                var runConfigArray = result.RunSettings.RunConfiguration[0];
+                                runConfigArray.MaxCpuCount = 0;
+                            }
+
+                            var builder = new xml2js.Builder();
+                            var runSettingsForParallel = builder.buildObject(result);
+                            saveToFile(runSettingsForParallel).then(function(fileName) {
+                                cleanUp(settingsFile);
+                                defer.resolve(fileName);
+                                return defer.promise;
+                            })
+                                .fail(function(err) {
+                                    tl.debug("Error occured while setting run in parallel. Continuing...");
+                                    tl.warning(err);
+                                    defer.resolve(settingsFile);
+                                });
+                        }
+                    });
+                })
+                    .fail(function(err) {
+                        tl.warning(err);
+                        tl.debug("Error occured while setting run in parallel. Continuing...");
+                        defer.resolve(settingsFile);
+                    });
+            }
+        }
+    }
+    else {
+        defer.resolve(settingsFile);
+    }
+
+    return defer.promise;
+}
+
+function saveToFile(fileContents: string): Q.Promise<string> {
```

Comment:
why this is not in the util file? save file is something generic, you should be passing file name and content